metrics/convert/convert.go

```diff
@@ -0,0 +1,135 @@
+// Package convert provides a way to use Counters, Histograms, or Gauges
+// as one of the other types
+package convert
```

Comment:
It feels like this is tailor-made to cloudwatch2, and I'm a little bit scared that folks will see it and think it's something they should be doing in their own code. Would you be OK with moving this underneath the internal/ dir?

Reply:
Will do!